src/strings/uthash.h

```diff
@@ -556,20 +557,95 @@ do {
     (head)=NULL;                                                                 \
   }                                                                              \
 } while(0)
-
-/* obtain a count of items in the hash */
-#define HASH_CNT(hh,head) ((head)?((head)->hh.tbl->num_items):0)
 #define GET_X_BITS(bucket_rand, log2_num_buckets) \
     ((bucket_rand) >> ((sizeof(bucket_rand) * 8) - (log2_num_buckets)))
+/* obtain a count of items in the hash */
+#define HASH_CNT(hh,head) ((head)?((head)->hh.tbl->num_items):0)
+
+MVM_STATIC_INLINE unsigned GET_X_BITS_FUNCT(MVMuint64 bucket_rand, unsigned log2_num_buckets) {
+    switch (log2_num_buckets) {
```

Comment:
> Also optimize GET_X_BITS by making a switch out of it. It doesn't seem
like this would be faster, but it is due to compiler optimizations.

Maybe it needs a comment here saying that it is done this way because it's faster? Otherwise someone may refactor it months later :)